flink-runtime/src/main/java/org/apache/flink/runtime/operators/resettable/SpillingResettableMutableObjectIterator.java

```diff
@@ -204,8 +204,16 @@ public void consumeAndCacheRemainingData() throws IOException {
 				} catch (IOException e) {
 					throw new RuntimeException("SpillingIterator: Error writing element to buffer.", e);
 				}
-				this.elementCount++;
+				increaseElementCount();
 			}
 		}
 	}
+
+	private void increaseElementCount() {
```

Comment:
We should avoid code duplication. Maybe it's better to add a utility function.

Reply:
Hi, @KarmaGYZ Thanks for your comment.

SpillingResettableIterator and SpillingResettableMutableObjectIterator  both implemented ResettableIterator, but they are two separate components. Although the logic of adding elementCount is somewhat similar, it is actually related to their respective implementation logic, so i think there is no need to share it.

Reply:
I don't think so. The logic of `increaseElementCount` in two classes is both check the numerical overflow and increase it. Am I understand it correctly?

Reply:
You can think of it this way, there may be many such scenes in reality, and you need to check the overflow, so whether all the places to check the overflow need to use the same code.

Reply:
Maybe you can think of it this way, there may be many such scenes in reality, and you need to check the overflow, so whether all the places to check the overflow need to use the same code.

And the point is that they are two separate components, and there is no need to reuse them for reuse.

Reply:
Extracting that logic to some `NumericUtil#safeIncrement` in `flink-core` could be a viable option.